### PR TITLE
fix(notification/rule): remove authID requirement

### PR DESCRIPTION
**Pull request by kelwang**
remove the authorization id requirement for creating rule. 
related to https://github.com/influxdata/influxdb/pull/14675

replace with the user id. 
Manually tested

remove the smtp notification rule

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by kelwang** [comment]
> I don't understand- I think we should just remove AuthorizationID and not add an OwnerID. @desa could you check this please?

ownerID is read only, frontend will never pass that in the post

File: notification/rule/smtp.go [hunk 1]
```diff
@@ -1,68 +0,0 @@
-package rule
```
Comment (goller) [inline notification/rule/smtp.go]: this should be in a different PR... not topical.

File: http/swagger.yml [hunk 2]
```diff
@@ -9057,8 +9057,8 @@ components:
         orgID:
           description: the ID of the organization that owns this notification rule.
           type: string
-        authorizationID:
-          description: The ID of the authorization used to create this notification rule.
+        ownerID:
```
Comment (goller) [inline http/swagger.yml]: @kelwang i think this should be a readonly field.  It should be sent via the POST/PATCH as owner is implied by the creator.

OwnerID would be gathered from the authorizer context

(context.Authorizer(ctx).GetUserID())

**Comment by chnn** [comment]
At some point I think it would be great for us to start defining separate schemas in the swagger spec for request and response bodies—it would clarify issues like whether or not we are supposed to send the `ownerID` when creating notification rules.

**Comment by goller** [comment]
@chnn agreed
